R/getGenomicTiles.R

```diff
@@ -0,0 +1,180 @@
+#' @title Get regions tiling a genome.
+#'
+#' @description Get sequential, potentially annotated regions of a fixed lengths
+#'    (tiles) along chromosomes of a genome.
+#'
+#' @author Michael Stadler
+#'
+#' @param genome The genome to work on. Either a \code{\link[BSgenome]{BSgenome}}
+#'     object, a \code{character} scalar with the name of an installed \code{\link[BSgenome]{BSgenome}}
+#'     or with a file path and name pointing to a fasta file with the genome sequence,
+#'     or a named \code{numeric} vector giving the names and lengths of chromosomes.
+#' @param tileWidth \code{numeric} scalar with the tile length.
+#' @param hasOverlap Named \code{list} with \code{\link[GenomicRanges]{GRanges}} object(s).
+#'     For each list element, a logical vector "X.hasOverlap" will be added to the
+#'     \code{mcols} of the result, with \code{TRUE} for each tile that overlaps
+#'     any region in that element. "X" is obtained from \code{names(hasOverlap)}.
+#' @param fracOverlap  Named \code{list} with \code{\link[GenomicRanges]{GRanges}} object(s).
+#'     For each list element, a numeric vector "X.fracOverlap" will be added to the
+#'     \code{mcols} of the result, with a value between 0 and 1 giving the fraction
+#'     of bases in a tile that overlaps with any region in that element. "X" is
+#'     obtained from \code{names(fracOverlap)}.
+#' @param numOverlap  Named \code{list} with \code{\link[GenomicRanges]{GRanges}} object(s).
+#'     For each list element, two numeric vectors "X.numOverlapWithin" and
+#'     "X.numOverlapAny" will be added to the \code{mcols} of the result, giving
+#'     then number of ranges in that element that are fully contained within
+#'     a tile, or that overlap with a tile in any way, respectively. "X" is
+#'     obtained from \code{names(numOverlap)}.
+#' @param nearest Named \code{list} with \code{\link[GenomicRanges]{GRanges}} object(s).
+#'     For each list element, two numeric vectors "X.nearestName" and
+#'     "X.nearestDistance" will be added to the \code{mcols} of the result, giving
+#'     the name and distance of nearest ranges in that element for each tile. "X" is
+#'     obtained from \code{names(nearest)}, and the values of "X.nearestName" from
+#'     \code{names(nearest$X)}.
+#'
+#' @details The last tile in each chromosome is dropped if it would be shorter
+#'     than \code{tileWidth}. Generated tiles are unstranded (\code{*}) and
+#'     therefore overlaps or searching for nearest neighbors are ignoring
+#'     strands of annotations (\code{ignore.strand=TRUE}). If multiple nearest
+#'     ranges are at the same distance from a tile, an arbitrary one is
+#'     reported in "X.nearestName".
```

Comment:
Yes, I think there might be cases where this could be of interest.

I am a bit reluctant to report several, because that would change the column type from a simple vector to a list-type, which can not be used e.g. in subsetting anymore: `distalGr <- gr[TSS.nearestDistance > 1000]`

Do you think it would be important to support that?

Reply:
As discussed we would currently keep the scalar return type by just reporting a single nearest element, and users have to call `nearest()` themselves if they need more information.